docs/user-guide/101-library.md

```diff
@@ -0,0 +1,109 @@
+`),
+	}
+	currentState := []byte(`                                                    
+routes:                                                                         
+  running:                                                                      
+  - destination: 0.0.0.0/0                                                      
+    next-hop-address: 192.168.100.1                                             
+    next-hop-interface: eth1                                                    
+    table-id: 254                                                               
+  - destination: 1.1.1.0/24                                                     
+    next-hop-address: 192.168.100.1                                             
+    next-hop-interface: eth1                                                    
+    table-id: 254                                                               
+interfaces:                                                                     
+- name: eth1                                                                    
+  type: ethernet                                                                
+  state: up                                                                     
+  mac-address: 00:00:5E:00:00:01                                                
+  ipv4:                                                                         
+    address:                                                                    
+    - ip: 10.244.0.1                                                            
+      prefix-length: 24                                                         
+    - ip: 169.254.1.0                                                           
+      prefix-length: 16                                                         
+    dhcp: true                                                                  
+    enabled: true                                                               
+`)
+	cachedState := types.CachedState{}
+	generatedState, err := nmpolicy.GenerateState(policySpec, currentState, cachedState)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(string(generatedState.DesiredState))
+}
+```
+
+It should output the following network state 
+```yaml
+interfaces:
+- bridge:
+    port:
+    - name: eth1
+  ipv4:
+    dhcp: true
+    enabled: true
+  name: br1
+  state: up
+  type: linux-bridge
+```
+
+Also the result from the ```capture``` evaluation will be returned at the
+```generateState.Cache.Capture``` [field](https://pkg.go.dev/github.com/nmstate/nmpolicy@v0.2.0/nmpolicy/types#CachedState)
+```yaml
+default-gw:
+  metaInfo:
+     time: "2021-12-15T13:45:40Z"
+  state:
+    routes:
+      running:
+      - destination: 0.0.0.0/0
+        next-hop-address: 192.168.100.1
+        next-hop-interface: eth1
+        table-id: 254
+```
+
+The ```generateState.Cache``` can be used as input to `nmpolicy.GenerateState`
```

Comment:
What is the motivation for using generateState.Cache over getting the current state?

Reply:
We use both, a policy capture expressions only fit on the current state before applying it, but if something on the desiredState has to be changed users can pass the previous current state so it can be re-calculated.